### PR TITLE
What's Here for Symbol

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: string.c [hunk 1]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ *             matches a given Regexp; +false+ otherwise.
+ * - #length, #size:: Returns the size of <tt>self.to_s</tt>.
+ * - #start_with?:: Returns +true+ if <tt>self.to_s</tt> starts with
+ *                  any of the given strings.
+ *
+ * === Methods for Comparing
+ *
+ * - {#<=>}[#method-i-3C-3D-3E]:: Returns -1, 0, or 1 as a given other string is smaller than, equal to, or larger than <tt>self.to_s</tt>.
+ * - {#==, #===}[#method-i-3D-3D]:: Returns +true+ if a given string
+ *                                  has the same content as <tt>self.to_s</tt>.
```
Comment (jeremyevans) [inline string.c:11314]: Symbols and strings are never considered equal.  `Returns true if the given symbol has the same content and encoding.`

File: string.c [hunk 2]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #casecmp:: Ignoring case, returns -1, 0, or 1 as a given
+ *              other string is smaller than, equal to, or larger than <tt>self.to_s</tt>.
```
Comment (jeremyevans) [inline string.c:11316]: Again, strings and symbols are not comparable.  `given other symbol` -> `given symbol`, and no mention of `to_s`.  Same for `casecmp?`

File: string.c [hunk 3]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #capitalize:: Returns <tt>self.to_s</tt> with the first character upcased
+ *                 and all other characters downcased.
+ * - #downcase:: Returns <tt>self.to_s</tt> with all characters downcased.
+ * - #inspect:: Returns the representation of +self+ as a symbol literal.
+ * - #name:: Returns the frozen string <tt>self.to_s</tt>.
+ * - #succ, #next:: Returns the string that is the successor to <tt>self.to_s</tt>.
+ * - #swapcase:: Returns <tt>self.to_s</tt> with all upcase characters downcased
+ *               and all downcase characters upcased.
+ * - #upcase:: Returns <tt>self.to_s</tt> with all characters upcased.
+ * - #to_proc:: Returns a Proc object which responds to the method given by sym.
```
Comment (jeremyevans) [inline string.c:11331]: `method given by sym` -> `method named by symbol`

File: string.c [hunk 4]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #to_s, #id2name:: Returns the name or string corresponding to +self+.
```
Comment (jeremyevans) [inline string.c:11332]: `name or string` -> `string ` (`name` or `string` sounds like they are two separate things)

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this.  Most of these descriptions need to be modified.  Strings and symbols are not comparable.  All cases where `self.to_s` is mentioned, it should be removed, in many cases replaced by `symbol`.

File: string.c [hunk 5]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ *            matches a given Regexp; +nil+ otherwise.
+ * - #match?:: Returns +true+ if <tt>self.to_s</tt>
+ *             matches a given Regexp; +false+ otherwise.
+ * - #length, #size:: Returns the size of <tt>self.to_s</tt>.
+ * - #start_with?:: Returns +true+ if <tt>self.to_s</tt> starts with
+ *                  any of the given strings.
+ *
+ * === Methods for Comparing
+ *
+ * - {#<=>}[#method-i-3C-3D-3E]:: Returns -1, 0, or 1 as a given other string is smaller than, equal to, or larger than <tt>self.to_s</tt>.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 1 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 2 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 3 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 4 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c [hunk 6]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #empty?:: Returns +true+ if +self.length+ is zero; +false+ otherwise.
+ * - #encoding:: Returns the Encoding object that represents the encoding
+ *               of symbol.
+ * - #end_with?:: Returns +true+ if symbol ends with
+ *                any of the given strings.
+ * - #match:: Returns a MatchData object if symbol
+ *            matches a given Regexp; +nil+ otherwise.
+ * - #match?:: Returns +true+ if symbol
+ *             matches a given Regexp; +false+ otherwise.
+ * - #length, #size:: Returns the size of symbol.
```
Comment (jeremyevans) [inline string.c:11306]: `size of` -> `number of characters in`.  Just to make it clear that this returns number of characters and not number of bytes.

File: string.c [hunk 7]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #casecmp?:: Returns +true+ if symbol is equal to a given symbol
+ *               after Unicode case folding; +false+ otherwise.
+ *
+ * === Methods for Converting
+ *
+ * - #capitalize:: Returns symbol with the first character upcased
+ *                 and all other characters downcased.
+ * - #downcase:: Returns symbol with all characters downcased.
+ * - #inspect:: Returns the representation of +self+ as a symbol literal.
+ * - #name:: Returns the frozen string symbol.
```
Comment (jeremyevans) [inline string.c:11326]: `Returns the frozen string corresponding to symbol.`

File: string.c [hunk 8]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #succ, #next:: Returns the string that is the successor to symbol.
```
Comment (jeremyevans) [inline string.c:11327]: `string` -> `symbol`

File: string.c [hunk 9]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #swapcase:: Returns symbol with all upcase characters downcased
+ *               and all downcase characters upcased.
+ * - #upcase:: Returns symbol with all characters upcased.
```
Comment (jeremyevans) [inline string.c:11330]: `upcase` is listed below, so this line should be removed (as it isn't in alphabetical order).

File: string.c [hunk 10]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ *              symbol is smaller than, equal to, or larger than symbol.
+ * - #casecmp?:: Returns +true+ if symbol is equal to a given symbol
+ *               after Unicode case folding; +false+ otherwise.
+ *
+ * === Methods for Converting
+ *
+ * - #capitalize:: Returns symbol with the first character upcased
+ *                 and all other characters downcased.
+ * - #downcase:: Returns symbol with all characters downcased.
+ * - #inspect:: Returns the representation of +self+ as a symbol literal.
```
Comment (jeremyevans) [inline string.c:11325]: `representation` -> `string representation`

File: string.c [hunk 11]
```diff
@@ -11273,6 +11273,66 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
...
+ * - #match?:: Returns +true+ if symbol
+ *             matches a given Regexp; +false+ otherwise.
+ * - #length, #size:: Returns the size of symbol.
+ * - #start_with?:: Returns +true+ if symbol starts with
+ *                  any of the given strings.
+ *
+ * === Methods for Comparing
+ *
+ * - {#<=>}[#method-i-3C-3D-3E]:: Returns -1, 0, or 1 as a given symbol is smaller than, equal to, or larger than symbol.
+ * - {#==, #===}[#method-i-3D-3D]:: Returns +true+ if a given string
```
Comment (jeremyevans) [inline string.c:11313]: `string` -> `symbol`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks better.  Still a few minor issues, please see inline comments.

File: string.c (hunk 6 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 7 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 8 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 9 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 10 above)
Comment (BurdetteLamar) [reply string.c]: Done.

File: string.c (hunk 11 above)
Comment (BurdetteLamar) [reply string.c]: Done.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Pull request merged**